internal/controller/instaslice_controller.go

```diff
@@ -678,15 +719,28 @@ func (l *RightToLeftPolicy) SetAllocationDetails(profileName string, newStart, s
 	return &inferencev1alpha1.AllocationRequest{}
 }
 
-func (r *InstasliceReconciler) removeInstasliceAllocation(ctx context.Context, instasliceName string, allocation *inferencev1alpha1.AllocationResult) error {
+func (r *InstasliceReconciler) removeInstasliceAllocation(ctx context.Context, allocation *inferencev1alpha1.AllocationResult, instaslice inferencev1alpha1.Instaslice, podUid types.UID) error {
+	log := logr.FromContext(ctx)
 	if allocation.AllocationStatus.AllocationStatusDaemonset == inferencev1alpha1.AllocationStatusDeleted {
-		err := utils.UpdateOrDeleteInstasliceAllocations(ctx, r.Client, instasliceName, nil, nil)
+		err := utils.UpdateOrDeleteInstasliceAllocations(ctx, r.Client, instaslice.Name, nil, nil)
 		if err != nil {
 			return err
 		}
 	}
+	allocRequest, exists := instaslice.Spec.PodAllocationRequests[podUid]
```

Comment:
removeInstaSliceallocation should only remove the allocation. I am confused, we deleted allocation and later we check for the same deleted allocation?